e2e/app/page/profile-page.ts

```diff
@@ -40,12 +47,64 @@ export default class ProfilePage extends AuthenticatedPage {
     }
   }
 
-  async getFirstName(): Promise<Textbox> {
-    return await Textbox.findByName(this.page, {name: LabelAlias.FirstName});
+  async getFirstNameInput(): Promise<Textbox> {
+    return Textbox.findByName(this.page, {dataTestId: DataTestIdAlias.FirstName});
+  }
+
+  async getLastNameInput(): Promise<Textbox> {
+    return Textbox.findByName(this.page, {dataTestId: DataTestIdAlias.LastName});
+  }
+
+  async getProfessionalUrlInput(): Promise<Textbox> {
+    return Textbox.findByName(this.page, {dataTestId: DataTestIdAlias.ProfessionalUrl});
+  }
+
+  async getResearchBackgroundTextarea(): Promise<Textarea> {
+    return Textarea.findByName(this.page, {normalizeSpace: LabelAlias.ResearchBackground});
+  }
+
+  async getAddress1Input(): Promise<Textbox> {
+    return Textbox.findByName(this.page, {dataTestId: DataTestIdAlias.Address1});
+  }
+
+  async getAddress2Input(): Promise<Textbox> {
+    return Textbox.findByName(this.page, {dataTestId: DataTestIdAlias.Address2});
   }
 
-  async getLastName(): Promise<Textbox> {
-    return await Textbox.findByName(this.page, {name: LabelAlias.LastName});
+  async getCityInput(): Promise<Textbox> {
+    return Textbox.findByName(this.page, {dataTestId: DataTestIdAlias.City});
   }
 
+  async getStateInput(): Promise<Textbox> {
+    return Textbox.findByName(this.page, {dataTestId: DataTestIdAlias.State});
+  }
+
+  async getZipCodeInput(): Promise<Textbox> {
+    return Textbox.findByName(this.page, {dataTestId: DataTestIdAlias.Zip});
+  }
+
+  async getCountryInput(): Promise<Textbox> {
+    return Textbox.findByName(this.page, {dataTestId: DataTestIdAlias.Country});
+  }
+
+  async getSaveProfileButton(): Promise<Button> {
+    return Button.findByName(this.page, {name: LabelAlias.SaveProfile});
+  }
+
+  // TODO generalize - promote to a Div Element?
+  async getDivWithText(text: string, options?: WaitForSelectorOptions): Promise<BaseElement> {
+    const selector = `//div[normalize-space(text())="${text}"]`;
+    const handle = await this.page.waitForXPath(selector, options);
+    return BaseElement.asBaseElement(this.page, handle);
+  }
+
+  async expectResearchPurposeErrorMissing(): Promise<void> {
+    const text = 'Current Research can\'t be blank';
+    await this.getDivWithText(text, {hidden: true});
+  }
+
+  async expectResearchPurposeErrorPresent(): Promise<void> {
```

Comment:
`await waitForText(this.page, text)` can be used to find texts on page.

Reply:
I want to have a generic check that works whether or not something is present - so I have changed this to something different